### ISSUE TITLE
Export promise status variables

### ISSUE THREAD

**Issue by piu130** [pburtchaell/redux-promise-middleware#143]
What do you say if we add the following lines [here](https://github.com/pburtchaell/redux-promise-middleware/blob/master/src/index.js#L2)

```javascript
export const PENDING = 'PENDING'
export const FULFILLED = 'FULFILLED'
export const REJECTED =  'REJECTED'

const defaultTypes = [PENDING, FULFILLED, REJECTED]; // remove ticks
```

Currently we have to listen to the actions with `${ACTION}_PENDING`, etc. I prefer to code with variables instead of strings. So I could add:

```javascript
import { PENDING } from 'redux-promise-middleware'
`${ACTION}_${PENDING}`
```

What do you think? I'd make a PR if you like it.

**Comment by pburtchaell** [comment]
I'm not opposed to exposing these strings as module exports. I also think it would be easy to maintain and import your own constants.

What do you think @tomatau?

**Comment by tomatau** [comment]
I like it :)

**Comment by pburtchaell** [comment]
Cool. Feel free to make a PR @Piu130. Please also include tests that evaluate the module exports! 👍

**Comment by pyros2097** [comment]
But according to https://github.com/pburtchaell/redux-promise-middleware/blob/master/docs/guides/custom-suffixes.md

You can specify custom suffixes other than PENDING..., then the exported constants PENDING won't be right/useful in you reducer and you might as well declare it as constants in your constants file. And btw type-to-reducer solves this problem very easily

**Comment by piu130** [comment]
With this `typeToReducer` module you hardcode `PENDING` every time. It's better you use the following code on [this](https://github.com/tomatau/type-to-reducer/blame/master/README.md#L53) line.
```js
import { PENDING } from 'redux-promise-middleware'
...
[PENDING]: () => ({ // Line 53
```
Then you have a dynamic `PENDING`.

We are in a discussion because of the naming in [this](https://github.com/pburtchaell/redux-promise-middleware/pull/144#issuecomment-287928884) PR. Maybe `DEFAULT_PENDING` would be better than `PENDING`?

**Comment by golinmarq** [comment]
@piu130 srsly? Do you prefer type DEFAULT_ instead two single quotes? That's insane! This issue should be close

**Comment by piu130** [comment]
@golinmarq It's not important if it's `DEFAULT_` or not. If you use `'_PENDING'` and you make a mistake eg. `'_PENDIN'` (missing E) it does not work and wont throw an error.

If you use it without single quote you can't make this mistake and your text editor will autocomplete it.

**Comment by golinmarq** [comment]
@piu130 sure, I get your point but, I think this could be done using other ways

**Comment by pburtchaell** [comment]
@golinmarq This feature is entirely optional. If you don't like it, you don't have to use it. 

You should also be considerate of @piu130 and other contributors on GitHub. This is a community of developers working together to make software better. Making comments like "srsly [...] this issue should be closed" and "I think this could be done using other ways" is not constructive feedback. 

If you'd like to formulate an actual argument or make a PR with your own solution, please feel free to, but irrationally bashing the work of others is not welcome.

**Comment by tomatau** [comment]
Totally agree with @pburtchaell on this.

Also, magic strings are a bad practice. Trimming characters from code before it's compiled is not really an optimal goal.

FWIW the PR has already been achieved "using other ways" and this PR will no doubt be closed once the PR has fixed the merge conflicts and is merged.

**Comment by pburtchaell** [comment]
By the way @piu130, I'm going to merge this once I have some time in the first week of May; I wanted to make a few changes before we merge. Sorry for the delay! I've been swamped with work.

**Comment by ShockiTV** [comment]
```
//elsewhere
export const FETCH_PROOFREADING_LIST = "profile/fetchProofreadingList";

export function pending (action) { return `${action}_PENDING`; }
export function success (action) { return `${action}_FULFILLED`; }
export function reject (action) { return `${action}_REJECTED`; }
```
```
//reducer
import * as a from "./actions";
import { pending, success, reject} from '../somewhere/helpers';

const proofreadingInitialState = {
  isInitialized: false,
  data: {
    error: null,
    payload: []
  }
};

export function proofreadingReducer(state = proofreadingInitialState, action) {
  switch (action.type) {
    case pending(a.FETCH_PROOFREADING_LIST):
      return proofreadingInitialState;
    case success(a.FETCH_PROOFREADING_LIST):
      return {
        ...state,
        isInitialized: true,
        data: {
          error: null,
          payload: action.payload
        }
      };
    case reject(a.FETCH_PROOFREADING_LIST):
      return {
        ...state,
        isInitialized: true,
        data: {
          error: action.error,
          payload: action.payload
        }
      };
    default:
      return state;
  }
}
```

**Comment by pburtchaell** [comment]
This has been resolved in #144 and released as `redux-promise-middleware@4.2.1`.
